generator/src/OperatorClassGenerator.php

```diff
@@ -61,83 +62,88 @@ public function createClass(GeneratorDefinition $definition, OperatorDefinition
         $namespace->addUse(Encode::class);
         $class->addConstant('ENCODE', new Literal('Encode::' . $operator->encode->name));
 
-        $constuctor = $class->addMethod('__construct');
+        $encodeNames = [];
+        $constructor = $class->addMethod('__construct');
         foreach ($operator->arguments as $argument) {
+            // Remove the leading $ from the argument name
+            $argName = ltrim($argument->name, '$');
+            $encodeNames[$argName] = $argument->noName ? null : $argument->name;
```

Comment:
How about adding methods to the argument class for these:
* `name` would return the name as-is
* `getParameterName` would return the name as used in factories, properties, etc.
* `getEncodedName` would return the name, or `null` if `noName` has been set.

This would avoid duplicating the logic in various generator classes.

Reply:
Added `$propName` property.